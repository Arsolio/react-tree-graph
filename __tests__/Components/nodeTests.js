import React from 'react';
import { shallow } from 'enzyme';

import Node from '../../src/components/node';

const defaultProps = {
	x: 1,
	y: 2,
	keyProp: '',
	labelProp: 'name',
	shape: 'circle',
	gProps: {
		className: 'test'
	},
	nodeProps: {},
	textProps: {},
	name: 'Test Node'
};

describe('<Node>', () => {
	test('renders circle correctly', () => {
		const tree = shallow(<Node {...defaultProps}/>);
		expect(tree).toMatchSnapshot();
	});

	test('renders circle correctly with custom radius', () => {
		const tree = shallow(<Node {...defaultProps} nodeProps={{ r: 10 }}/>);
		expect(tree).toMatchSnapshot();
	});

	test('renders rect correctly', () => {
		const tree = shallow(<Node {...defaultProps} shape="rect"/>);
		expect(tree).toMatchSnapshot();
	});

	test('renders rect correctly with custom size', () => {
		const tree = shallow(<Node {...defaultProps} shape="rect" nodeProps={{ height: 20, width: 30 }}/>);
		expect(tree).toMatchSnapshot();
	});

	test('renders image correctly', () => {
		const tree = shallow(<Node {...defaultProps} shape="image" nodeProps={{ href: 'http://example.com' }}/>);
		expect(tree).toMatchSnapshot();
	});

	test('renders image correctly with custom size', () => {
		const tree = shallow(
			<Node {...defaultProps} shape="image" nodeProps={{ href: 'http://example.com', height: 20, width: 30 }}/>
		);
		expect(tree).toMatchSnapshot();
	});

	test('click event has correct parameters', () => {
		const clickMock = jest.fn();
		const event = {};

		const props = {
			keyProp: 'id',
			gProps: {
				onClick: clickMock
			},
			id: 'testKey'
		};

		const tree = shallow(<Node {...defaultProps} {...props}/>);
		tree.find('g').simulate('click', event);

		expect(clickMock).toHaveBeenCalledTimes(1);
		expect(clickMock).toHaveBeenCalledWith(event, 'testKey');
	});

	test('right click event has correct parameters', () => {
		const rightClickMock = jest.fn();
		const event = {};

		const props = {
			keyProp: 'id',
			gProps: {
				onContextMenu: rightClickMock
			},
			id: 'testKey'
		};

		const tree = shallow(<Node {...defaultProps} {...props}/>);
		tree.find('g').simulate('contextmenu', event);

		expect(rightClickMock).toHaveBeenCalledTimes(1);
		expect(rightClickMock).toHaveBeenCalledWith(event, 'testKey');
	});


	test('clicking with no prop handler does nothing', () => {
		const props = {
			keyProp: 'id',
			id: 'testKey'
		};

		const tree = shallow(<Node {...defaultProps} {...props}/>);
		tree.find('g').simulate('click');
	});

	test('htmlProps applied to all elements', () => {
		const props = {
			gProps: {
				className: 'g'
			},
			nodeProps: {
				className: 'circle'
			},
			textProps: {
				className: 'text'
			}
		};

		const tree = shallow(<Node {...defaultProps} {...props}/>);
		expect(tree).toMatchSnapshot();
	});
});