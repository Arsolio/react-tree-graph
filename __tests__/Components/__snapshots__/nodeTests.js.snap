// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Node> htmlProps applied to all elements 1`] = `
<g
  className="g"
  transform="translate(2, 1)"
>
  <circle
    className="circle"
    r={5}
  />
  <text
    className="text"
    dx={5.5}
    dy={5}
  >
    Test Node
  </text>
</g>
`;

exports[`<Node> renders circle correctly 1`] = `
<g
  className="test"
  transform="translate(2, 1)"
>
  <circle
    r={5}
  />
  <text
    dx={5.5}
    dy={5}
  >
    Test Node
  </text>
</g>
`;

exports[`<Node> renders circle correctly with custom radius 1`] = `
<g
  className="test"
  transform="translate(2, 1)"
>
  <circle
    r={10}
  />
  <text
    dx={10.5}
    dy={5}
  >
    Test Node
  </text>
</g>
`;

exports[`<Node> renders custom label correctly 1`] = `
<g
  className="test"
  transform="translate(2, 1)"
>
  <circle
    r={5}
  />
  <g
    transform="translate(5.5, 5)"
  >
    <circle
      r="5"
    />
  </g>
</g>
`;

exports[`<Node> renders image correctly 1`] = `
<g
  className="test"
  transform="translate(2, 1)"
>
  <image
    height={10}
    href="http://example.com"
    width={10}
    x={-5}
    y={-5}
  />
  <text
    dx={5.5}
    dy={5}
  >
    Test Node
  </text>
</g>
`;

exports[`<Node> renders image correctly with custom size 1`] = `
<g
  className="test"
  transform="translate(2, 1)"
>
  <image
    height={20}
    href="http://example.com"
    width={30}
    x={-15}
    y={-10}
  />
  <text
    dx={15.5}
    dy={5}
  >
    Test Node
  </text>
</g>
`;

exports[`<Node> renders rect correctly 1`] = `
<g
  className="test"
  transform="translate(2, 1)"
>
  <rect
    height={10}
    width={10}
    x={-5}
    y={-5}
  />
  <text
    dx={5.5}
    dy={5}
  >
    Test Node
  </text>
</g>
`;

exports[`<Node> renders rect correctly with custom size 1`] = `
<g
  className="test"
  transform="translate(2, 1)"
>
  <rect
    height={20}
    width={30}
    x={-15}
    y={-10}
  />
  <text
    dx={15.5}
    dy={5}
  >
    Test Node
  </text>
</g>
`;
